### PR TITLE
43807: fix rms post all gather kernel bug

### LINKED ISSUE #43807: ttnn.rms_norm_pre_all_gather / ttnn.rms_norm_post_all_gather produce garbage output when invoked with use_2d_core_grid=True
### Summary

  ttnn.rms_norm_pre_all_gather / ttnn.rms_norm_post_all_gather produce garbage output when invoked with use_2d_core_grid=True. With the flag disabled, the same call returns numerically correct results.

  ### Where it shows up
  Originally observed in the LLaMA 70B Galaxy (6u, 8x4 mesh) at models/demos/llama3_70b_galaxy/tt/llama_ccl.py::tt_distributed_rmsnorm. We enable use_2d_grid whenever inp.shape[-2] == 128 (decode shape).

  ### Repro
  Single-device reproducer added at tests/ttnn/unit_tests/operations/fused/test_rmsnorm_2d_core_grid_single_device.py. It simulates the multi-device flow by chunking the input along the
  hidden dim, running rms_norm_pre_all_gather per chunk, concatenating the stats (stand-in for the all-gather), then running rms_norm_post_all_gather per chunk. The use_2d_core_grid flag
  is parametrized so the failing and passing configs run side by side.

  Config that reproduces:
  - seq_len=128, hidden_dim_total=8192, num_simulated_devices=4
  - dtype=bfloat16, HiFi4, fp32_dest_acc_en=True
  - use_2d_core_grid=True → PCC < 0.99 (fails)
  - use_2d_core_grid=False → passes

  ### Run

  Checkout `aling/llama-rmsnorm`

  pytest tests/ttnn/unit_tests/operations/fused/test_rmsnorm_2d_core_grid_single_device.py

  Expected vs actual

  Expected: PCC ≥ 0.99 against torch.nn.RMSNorm for both flag values.
  Actual: use_2d_core_grid=True produces numerically incorrect output; False matches reference.

### PR DESCRIPTION
**Pull request by bbradelTT**
### PR Category

Closes #43807 

2d core grid path for rms norm distributed was failing. tile counts were not set up correctly for that case. This PR fixes that and adds a test.

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:bbradel-43807_rms_post_fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:bbradel-43807_rms_post_fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:bbradel-43807_rms_post_fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:bbradel-43807_rms_post_fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:bbradel-43807_rms_post_fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:bbradel-43807_rms_post_fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:bbradel-43807_rms_post_fix)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:bbradel-43807_rms_post_fix)

### CODE REVIEW

File: tests/ttnn/nightly/unit_tests/operations/fused/test_distributed_rmsnorm_allgather.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        for chunk in input_chunks
+    ]
+    tt_weights = [
+        ttnn.from_torch(
+            w.reshape(1, 1, 1, hidden_per_dev),
+            dtype=ttnn.bfloat16,
+            device=device,
+            layout=ttnn.TILE_LAYOUT,
+            memory_config=ttnn.DRAM_MEMORY_CONFIG,
+        )
```
[the remark this answers is not in the document]
Comment (bbradelTT) [reply tests/ttnn/nightly/unit_tests/operations/fused/test_distributed_rmsnorm_allgather.py:65]: Switching this would obscure the connection to the production code path for the original bug for the LLaMA 70B Galaxy tt_distributed_rmsnorm flow.

**Review by vsureshTT — APPROVED** [review]
